backend/models/group.js

```diff
@@ -406,271 +430,90 @@ groupSchema.statics.listGroups = function (dbCol, queryParams, branch, revId, id
+				result.objects = sharedIdObjects;
+				return result;
+			})
+		);
 	});
-};
-
-groupSchema.statics.updateIssueId = function (dbCol, uid, issueId) {
 
-	if ("[object String]" === Object.prototype.toString.call(uid)) {
-		uid = utils.stringToUUID(uid);
-	}
+	const sharedIdGroups = await Promise.all(sharedIdConversionPromises);
 
-	return this.findOne(dbCol, { _id: uid }).then(group => {
-		if (group) {
-			const issueIdData = {
-				issue_id: issueId
-			};
-
-			return group.updateAttrs(dbCol, issueIdData);
-		} else {
-			return Promise.reject(responseCodes.GROUP_NOT_FOUND);
-		}
+	sharedIdGroups.forEach((group, i) => {
+		sharedIdGroups[i] = clean(group);
 	});
-};
 
-// Group Update with Event
-groupSchema.methods.updateGroup = function (dbCol, sessionId, data, user = "", branch = "master", rid = null) {
-	return this.updateAttrs(dbCol, _.cloneDeep(data), user).then((savedGroup) => {
-		return getObjectIds(dbCol, this, branch, rid, true, false).then((objects) => {
-			savedGroup.objects = objects;
-			ChatEvent.groupChanged(sessionId, dbCol.account, dbCol.model, savedGroup);
-			return savedGroup;
-		});
-	});
+	return sharedIdGroups;
 };
 
-groupSchema.methods.updateAttrs = function (dbCol, data, user) {
-	return this.getObjectsArrayAsIfcGuids(data, false).then(convertedObjects => {
-		const toUpdate = {};
-		const toUnset = {};
-		const fieldsCanBeUpdated = ["description", "name", "rules", "objects", "color", "transformation", "issue_id", "risk_id"];
-
-		let typeCorrect = !(data.rules && data.objects);
-		typeCorrect && fieldsCanBeUpdated.forEach((key) => {
-			if (data[key]) {
-				if (Object.prototype.toString.call(data[key]) === fieldTypes[key]) {
-					if (key === "objects" && data.objects) {
-						toUpdate.objects = cleanEmbeddedObject(key, convertedObjects);
-						toUnset.rules = 1;
-						this.rules = undefined;
-					} else if (key === "color" || key === "transformation") {
-						toUpdate[key] = data[key].map((c) => parseInt(c, 10));
-					} else {
-						if (key === "rules"
-							&& data.rules
-							&& !checkRulesValidity(data.rules)) {
-							typeCorrect = false;
-							toUnset.objects = 1;
-							this.objects = undefined;
-						}
-
-						toUpdate[key] = cleanEmbeddedObject(key, data[key]);
-					}
-					this[key] = toUpdate[key];
+Group.update = async function (account, model, branch = "master", revId = null, sessionId, user = "", groupId, data) {
+	// return updateAttrs(account, model, groupId, branch, rid, _.cloneDeep(data), user).then((savedGroup) => {
+	const group = await Group.findByUID(account, model, branch, revId, groupId);
+
+	const convertedObjects = await getObjectsArrayAsIfcGuids(data, false);
+	const toUpdate = {};
+	const toUnset = {};
+	const fieldsCanBeUpdated = ["description", "name", "rules", "objects", "color", "transformation", "issue_id", "risk_id"];
+
+	let typeCorrect = !(data.rules && data.objects);
+
+	typeCorrect && fieldsCanBeUpdated.forEach((key) => {
```

Comment:
Also similar to create, we should just ditch this approach and turn it into a switch?